scientific_library/tvb/simulator/models/infinite_theta.py

```diff
@@ -0,0 +1,871 @@
+    we express the system dynamics in terms of two state variables :math:`r` and :math:`V` representing the average firing rate and the average membrane potential of our QIF neurons. 
+    The conductance variable and its derivative are :math:`g` and :math:`q`.
+
+    The equations of the model read
+    .. math::
+            \dot{r} &= Delta/pi + 2 V r - g r^2),
+            \dot{V} &= V^2 - pi^2 r^2 + eta + (v_syn - V) g ,
+            \dot{g} &= alpha q , \\
+            \dot{q} &= alpha (k pi r - g - 2 q)
+            
+    """
+    
+    # Define traited attributes for this model, these represent possible kwargs.
+
+    Delta = NArray(
+        label=r":math:`\Delta`",
+        default=numpy.array([0.5]),
+        domain=Range(lo=0.0, hi=10.0, step=0.01),
+        doc="""Half-width of heterogeneous noise distribution""",
+    )
+    
+    alpha = NArray(
+        label=r":math:`\alpha`",
+        default=numpy.array([0.95]),
+        domain=Range(lo=0.0, hi=10.0, step=0.01),
+        doc="""Parameter of the alpha-function""",
+    )
+    
+    v_syn = NArray(
+        label=":math:`v_syn`",
+        default=numpy.array([-10.0]),
+        domain=Range(lo=-20.0, hi=0.0, step=0.01),
+        doc="""QIF membrane reversal potential""",
+    )
+    
+    k = NArray(
+        label=":math:`k`",
+        default=numpy.array([1.]),
+        domain=Range(lo=0., hi=5.0, step=0.01),
+        doc="""Local coupling strength""",
+    )
+
+    eta = NArray(
+        label=r":math:`\eta`",
+        default=numpy.array([20.0]),
+        domain=Range(lo=-10.0, hi=10.0, step=0.0001),
+        doc="""Constant parameter to scale the rate of feedback from the
+            firing rate variable to itself""",
+    )
+
+
+    #Informational attribute, used for phase-plane and initial()
+    state_variable_range = Final(
+        label="State Variable ranges [lo, hi]",
+        default={"r": numpy.array([1./numpy.pi, 1./numpy.pi]),
+                "V": numpy.array([0., 0.]),
+                "g": numpy.array([0., 0.]),
+                "q": numpy.array([0., 0.])},
+        doc="""Expected ranges of the state variables for initial condition generation and phase plane setup.""",
+    )
+
+    state_variable_boundaries = Final(
+            label="State Variable boundaries [lo, hi]",
+            default={
+                "r": numpy.array([0.0, numpy.inf])
+            },
+    )
+    
+    variables_of_interest = List(
+            of=str,
+            label="Variables or quantities available to Monitors",
+            choices=("r", "V", "g", "q"),
+            default=("r", "V"),
+            doc="The quantities of interest for monitoring for the Infinite QIF 2D oscillator.",
+    )
+
+
+    state_variables = ('r', 'V', 'g', 'q')
+    _nvar = 4
+    # Cvar is the coupling variable. 
+    cvar = numpy.array([0,1,2,3], dtype=numpy.int32)
+    
+    def dfun(self, state_variables, coupling, local_coupling=0.0):
+        r = state_variables[0, :]
```

Comment:
`r, V, g, q = state_variables` is a little nicer, also ensures `state_variables.shape[0]==self._nvar`